### ISSUE TITLE
quota step error

### ISSUE THREAD

**Issue by heyitsaakash** [aitalian/apigeex-challenge-lab#1]
Hi did u get passed - Please add the 'Q-Enforce Quota' Quota policy with the correct configuration to the proxy endpoint preflow and deploy the API proxy. I have passed the Verify API key check but this quota step is still showing error.

I used same configuration as you have provided. Is it still an issue in qwicklabs

`<?xml version="1.0" encoding="UTF-8" standalone="yes"?>
<Quota async="false" continueOnError="false" enabled="true" name="Q-EnforceQuota" type="calendar">
    <DisplayName>Q-EnforceQuota</DisplayName>
    <StartTime>2022-07-01 12:00:00</StartTime>
    <UseQuotaConfigInAPIProduct stepName="VAK-VerifyKey">
        <DefaultConfig>
            <Allow countRef="verifyapikey.VAK-VerifyKey.apiproduct.developer.quota.limit">5</Allow>
            <Interval ref="verifyapikey.VAK-VerifyKey.apiproduct.developer.quota.interval">1</Interval>
            <TimeUnit ref="verifyapikey.VAK-VerifyKey.apiproduct.developer.quota.timeunit">hour</TimeUnit>
        </DefaultConfig>
    </UseQuotaConfigInAPIProduct>
    <Distributed>true</Distributed>
    <Synchronous>true</Synchronous>
</Quota>`

thanks

**Comment by aitalian** [comment]
Task 2 now passes, however Task 3 still has an error on the Quota step (score: 15/20; overall 95/100). I have a support ticket in with Qwiklabs.

**Comment by venkatesulua1990** [comment]
Below configuration will work 
<Quota continueOnError="false" enabled="true" name="Q-EnforceQuota" type="calendar">
    <Identifier ref="client_id"/>
    <UseQuotaConfigInAPIProduct stepName="VAK-VerifyKey">
        <DefaultConfig>
            <Allow>5</Allow>
            <Interval>1</Interval>
            <TimeUnit>hour</TimeUnit>
        </DefaultConfig>
    </UseQuotaConfigInAPIProduct>
    <Distributed>true</Distributed>
    <Synchronous>true</Synchronous>
    <StartTime>2021-01-01 00:00:00</StartTime>
</Quota>

**Comment by venkatesulua1990** [comment]
Please use Task 4. Enforce a quota from your lab and it worked for me.

>

> <Quota continueOnError="false" enabled="true" name="Q-EnforceQuota" type="calendar">
    <Identifier ref="client_id"/>
    <UseQuotaConfigInAPIProduct stepName="VAK-VerifyKey">
        <DefaultConfig>
            <Allow>5</Allow>
            <Interval>1</Interval>
            <TimeUnit>hour</TimeUnit>
        </DefaultConfig>
    </UseQuotaConfigInAPIProduct>
    <Distributed>true</Distributed>
    <Synchronous>true</Synchronous>
    <StartTime>2021-01-01 00:00:00</StartTime>
</Quota>

**Comment by ajayprasath8600** [comment]
Able to complete 100% using this @venkatesulua1990  ??

**Comment by venkatesulua1990** [comment]
Yes 100% complete, Identifier elements in the Quota policy to define multiple, unique counters in a single policy. By using these elements, a single policy can maintain different counters based on the app making the request, the app developer making the request, a client ID or other client identifier, and more
